gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GcsioTrackingHttpRequestInitializer.java

```diff
@@ -0,0 +1,75 @@
+package com.google.cloud.hadoop.gcsio;
+/*
+ * Copyright 2019 Google LLC. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import com.google.api.client.http.HttpExecuteInterceptor;
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.http.HttpRequestInitializer;
+import com.google.common.collect.ImmutableList;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class GcsioTrackingHttpRequestInitializer implements HttpRequestInitializer {
+
+  private final HttpRequestInitializer delegate;
```

Comment:
Are there any usages of `delegate` in the context ?

Reply:
we are using the delegate to initialize